File: VibeCheckAI-ViewReact/src/app/socialMediaAnalysis/youtube/service/YouTubeAnalysisResponse.dto.ts
import {EmotionsEnum} from 'App/socialMediaAnalysis/enum/EmotionsEnum.enum';
import {SentimentEnum} from 'App/socialMediaAnalysis/enum/SentimentEnum.enum';
import {YouTubeChannelInformationDTO} from './YouTubeChannelInformation.dto';
import {YouTubeVideoInformationDTO} from './YouTubeVideoInformation.dto';

export interface YouTubeAnalysisResponseDTO {
    videoStatisticsInfo: YouTubeVideoInformationDTO,
    channelInfo: YouTubeChannelInformationDTO,
    emotions: Map<EmotionsEnum, number>,
    sentiments: Map<SentimentEnum, number>,
    toxicity: number
}

File: VibeCheckAI-Logic/src/main/java/com/example/vibecheckai/youtube/dto/YouTubeAnalysisResponseDTO.java
package com.example.vibecheckai.youtube.dto;

import com.example.vibecheckai.enums.EmotionsEnum;
import com.example.vibecheckai.enums.SentimentEnum;

import java.util.Map;

public record YouTubeAnalysisResponseDTO(
    YouTubeAnalysisVideoStatistics videoStatisticsInfo,
    YouTubeAnalysisChannelInfoDTO channelInfo,
    Map<EmotionsEnum, Integer> emotions,
    Map<SentimentEnum, Integer> sentiments,
    Float toxicity
) {

}


